File: day6/day6.js
import { input, testInput } from "./input.js";

const part1 = () => {
  console.log("Welcome To Day 6");
  // let howLongItTook = 0;
  const splitInput = input.split("");

  let runningArray = [];
  let answer;

  for (let i = 0; i < splitInput.length - 3; i++) {
    runningArray = [];
    let first = splitInput[i];
    let second = splitInput[i + 1];
    let third = splitInput[i + 2];
    let fourth = splitInput[i + 3];
    if (first != second) {
      runningArray.push(first);
      runningArray.push(second);
      if (!runningArray.includes(third)) {
        runningArray.push(third);
        if (!runningArray.includes(fourth)) {
          answer = i + 4;
          break;
        }
      }
    }
  }
  console.log(answer);
};

part1();

File: 2025/day6/day6.py
from input import input_data, test_input

def calculate_total(numbers, sign):
    """Calculate total of numbers using the given operator."""
    if sign == '+':
        return sum(numbers)
    elif sign == '*':
        result = 1
        for num in numbers:
            result *= num
        return result
    else:
        raise ValueError(f"Unknown operator: {sign}")

def part1(data_source):
    print('Welcome To Day 6')
    grand_total = 0
    lines = data_source.strip().split('\n')
    
    # Parse each line into values, splitting by whitespace
    rows = [line.split() for line in lines]
    
    # Transpose rows into columns
    columns = list(zip(*rows))
    
    # Process each column: convert numbers to int, keep last element as operator
    processed_columns = []
    for col in columns:
        numbers = [int(x) for x in col[:-1]]  # All but last are numbers
        operator = col[-1]  # Last element is the operator
        processed_columns.append((numbers, operator))
    
    # Calculate and print results for each column
    for i, (nums, op) in enumerate(processed_columns):
        total = calculate_total(nums, op)
        grand_total += total
    print(f"Grand Total: {grand_total}")
    pass

    # def part2(data_source):
    #     print('Welcome To Day 6: Part 2')
    #     lines = data_source.strip().split('\n')
    #     
    #     # Your solution here
    #     pass

if __name__ == '__main__':
    # part1(test_input)
    part1(input_data)
    
    # part2(test_input)
    # part2(input_data)

